File: app/src/main/java/com/example/myanimedata/api/MangaResponse.java
package com.example.myanimedata.api;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class MangaResponse {
    @SerializedName("data")
    private final List<MangaResult> mangaResults = new ArrayList<>();

    public List<MangaResult> getMangaResults() {
        return mangaResults;
    }
}

File: app/src/main/java/com/example/myanimedata/api/MangaResponse.kt
package com.example.myanimedata.api

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class MangaResponse (
    @field:SerializedName("data") val mangaResults: ArrayList<MangaResult>? = null
) : Parcelable